internal/mesh/internal/mappers/sidecarproxymapper/proxy_configuration_mapper_test.go

```diff
@@ -0,0 +1,73 @@
+package sidecarproxymapper
+
+import (
+	"context"
+	"testing"
+
+	svctest "github.com/hashicorp/consul/agent/grpc-external/services/resource/testing"
+	"github.com/hashicorp/consul/internal/catalog"
+	"github.com/hashicorp/consul/internal/controller"
+	"github.com/hashicorp/consul/internal/mesh/internal/cache/sidecarproxycache"
+	"github.com/hashicorp/consul/internal/mesh/internal/types"
+	"github.com/hashicorp/consul/internal/resource"
+	"github.com/hashicorp/consul/internal/resource/resourcetest"
+	pbcatalog "github.com/hashicorp/consul/proto-public/pbcatalog/v1alpha1"
+	pbmesh "github.com/hashicorp/consul/proto-public/pbmesh/v1alpha1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+	"github.com/hashicorp/consul/proto/private/prototest"
+	"github.com/stretchr/testify/require"
+)
+
+func TestProxyConfigurationMapper(t *testing.T) {
```

Comment:
Given that the mapper doesn't delete it, I don't think we need to test it there. We usually don't care about deletes in the mapper because the Reconcile should take care of it. This is tested in the data_fetcher.go (`TestFetcher_FetchAndMergeProxyConfigurations`)